src/lib.rs

```diff
@@ -28,21 +38,13 @@ impl Build {
             .context("cannot get package's metadata")?;
 
         log::trace!("Build: Initialize build process");
-        let mut build_process = process::Command::new("cargo");
-        build_process
-            .current_dir(&metadata.workspace_root)
```

Comment:
you need this too... 🤔 but it's annoying because the user won't be able to override it

Reply:
I was thinking about something similar to `exclude_path` and `exclude_workspace_path`

Reply:
So here is something we can do:

1. We need to provide the user an easy access to the `Metadata` (from cargo_metadata). For that we can create a public root level function that returns it. To avoid running and parsing multiple times the metadata and to avoid adding dependencies, we gonna need to use lazy_static which is already added by structopt. Here is the idea, try run it to see how the object get constructed only once and lazily https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5e593e9e9dfdfb2b422f14d7ba0912ed (dont use once_cell since we dont have it in our dependencies)
2. We need to add a public field `run_in_workspace_root: bool` to `Build`. Make it `true` by default. Then, in the `execute` function, if this is true we do override the current_dir of the command by the workspace root. Make this field skipped by structopt, it is not intended for the end user.
3. Make sure you use your pub function created in point 1 everywhere you need to get the metadata